website/docs/packages/concepts/README.md

```diff
@@ -0,0 +1,78 @@
+# Concepts
+
+Reusability is a pivotal aspect of application development, offering benefits such as accelerated development, centralized business logic, and the introduction of standardization into applications.
+
+## Modules in Appsmith
+
+Modules in Appsmith are powerful abstractions that facilitate the reuse and organization of code, promoting scalability and maintainability in application development.
+
+* **Reusability**: Modules allow developers to create a shared library of reusable components, widgets, and queries that can be used across multiple applications.
+
+* **Track and Manage Updates:**: Any modification made to a module is immediately reflected in every application where the module is used, ensuring consistency and easy maintenance.
+
+* **Composability**: Modules enable bidirectional communication between the parent app and the module through data inputs and query inputs, enhancing composability.
+
+* **Organizational Efficiency**: Modules focus on keeping apps and building blocks organized, making it easier to update, faster to build, and less prone to debugging issues.
+
+<ZoomImage
+  src="/img/modules-con.png" 
+  alt=""
+  caption=""
+/>
+
+### Query Modules
+
+Query Modules in Appsmith are specialized modules designed to encapsulate and reuse database queries across multiple applications within the same workspace. 
+
+When working with queries, there might be scenarios where you need to change query parameters dynamically. Query Modules facilitate this by providing `inputs` that can be linked to widgets or app-level parameters. These inputs act as dynamic variables, enabling developers to modify query parameters based on user interactions or changing application requirements. 
+
+* Query Modules require a dedicated datasource creation for context-specific data retrieval. Datasources created at the app level won't work within Query Modules.
```

Comment:
Didn't understand this part. Why aren't modules able to access the same datasources? If the datasource is in the same workspace it will be accessible in modules.